code/modules/food_and_drinks/drinks/drinks/drinkingglass.dm

```diff
@@ -16,517 +16,23 @@
-				desc = "Kahlua, Irish Cream, and cognac. You will get bombed."
-			if("toxinsspecial")
-				icon_state = "toxinsspecialglass"
-				name = "Toxins Special"
-				desc = "Whoah, this thing is on FIRE!"
-			if("chocolatepudding")
-				icon_state = "chocolatepudding"
-				name = "Chocolate Pudding"
-				desc = "Tasty."
-			if("vanillapudding")
-				icon_state = "vanillapudding"
-				name = "Vanilla Pudding"
-				desc = "Tasty."
-			if("cherryshake")
-				icon_state = "cherryshake"
-				name = "Cherry Shake"
-				desc = "A cherry flavored milkshake."
-			if("bluecherryshake")
-				icon_state = "bluecherryshake"
-				name = "Blue Cherry Shake"
-				desc = "An exotic blue milkshake."
-			if("drunkenblumpkin")
-				icon_state = "drunkenblumpkin"
-				name = "Drunken Blumpkin"
-				desc = "A drink for the drunks."
-			if("pumpkin_latte")
-				icon_state = "pumpkin_latte"
-				name = "Pumpkin Latte"
-				desc = "A mix of coffee and pumpkin juice."
-			if("gibbfloats")
-				icon_state = "gibbfloats"
-				name = "Gibbfloat"
-				desc = "Dr. Gibb with ice cream on top."
-			if("whiskey_sour")
-				icon_state = "whiskey_sour"
-				name = "Whiskey Sour"
-				desc = "Lemon juice mixed with whiskey and a dash of sugar. Surprisingly satisfying."
-			if("fetching_fizz")
-				icon_state = "fetching_fizz"
-				name = "Fetching Fizz"
-				desc = "Induces magnetism in the imbiber. Started as a barroom prank but evolved to become popular with miners and scrappers. Metallic aftertaste."
-			if("hearty_punch")
-				icon_state = "hearty_punch"
-				name = "Hearty Punch"
-				desc = "Aromatic beverage served piping hot. According to folk tales it can almost wake the dead."
-			if("absinthe")
-				icon_state = "absinthe"
-				name = "glass of absinthe"
-				desc = "It's as strong as it smells."
-			if("bacchus_blessing")
-				icon_state = "glass_brown2"
-				name = "Bacchus' Blessing"
-				desc = "You didn't think it was possible for a liquid to be so utterly revolting. Are you sure about this...?"
-			if("arnold_palmer")
-				icon_state = "arnold_palmer"
-				name = "Arnold Palmer"
-				desc = "You feel like taking a few golf swings after a few swigs of this."
-			if("hcider")
-				icon_state = "whiskeyglass"
-				name = "Hard Cider"
-				desc = "Tastes like autumn."
-			if("triple_citrus")
-				icon_state = "triplecitrus" //needs own sprite mine are trash
-				name = "glass of triple citrus"
-				desc = "A mixture of citrus juices. Tangy, yet smooth."
-			if("grappa")
-				icon_state = "grappa"
-				name = "glass of grappa"
-				desc = "A fine drink originally made to prevent waste by using the leftovers from winemaking."
-			if("eggnog")
-				icon_state = "glass_yellow"
-				name = "Eggnog"
-				desc = "For enjoying the most wonderful time of the year."
+		var/datum/reagent/largest_reagent = reagents.get_master_reagent()
+		if(largest_reagent)
+			if(largest_reagent.glass_name)
+				name = "[largest_reagent.glass_name]"
+
+				if(largest_reagent.glass_desc)
```

Comment:
This can be made a lot simpler:

```
/obj/item/weapon/reagent_containers/food/drinks/drinkingglass/on_reagent_change()
	if(reagents.reagent_list.len)
		var/datum/reagent/R = reagents.get_master_reagent()
		icon_state = R.drink_icon
		name = R.drink_name
		desc = R.drink_desc
	else
		icon_state = "glass_empty"
		name = "glass"
		desc = "Your standard drinking glass."
```